src/main/java/io/kestra/plugin/microsoft365/outlook/Get.java

```diff
@@ -0,0 +1,179 @@
+        description = "Uniques identifier of rhe email message to retrieve"
+    )
+    @NotNull
+    private Property<String> messageId;
+
+    @Schema(
+        title = "Include attachments",
+        description = "Whether to include attachment information in the response"
+    )
+    @Builder.Default
+    private Property<Boolean> includeAttachments = Property.ofValue(false);
+
+    @Schema(
+        title = "User email",
+        description = "Email address of the user whose mailbox to access (optional, uses authenticated user if not specified)"
+    )
+    private Property<String> userEmail;
+
+    @Override
+    public Output run (RunContext runContext) throws Exception {
+        Logger logger = runContext.logger();
+
+        GraphServiceClient graphClient = this.createGraphClient(runContext);
+
+        String rMsgId = runContext.render(messageId).as(String.class).orElseThrow();
+        String rUser = userEmail != null ? runContext.render(userEmail).as(String.class).orElse(null) : null;
+        Boolean rIncludeAttachment = runContext.render(includeAttachments).as(Boolean.class).orElse(false);
+
+        if (rUser != null && !rUser.matches("^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$")) {
+            throw new IllegalArgumentException("Invalid email format for userEmail: " + rUser);
+        }
+
+        logger.info("Retrieving message '{}' for user: {}", rMsgId, rUser != null ? rUser : "current user");
+
+        // Get the message
+        Message message = GraphMailUtils.fetchMessage(graphClient, rUser, rMsgId);
+
+        if (message == null) {
+            throw new IllegalStateException("Failed to retrieve message from Microsoft Graph API");
+        }
+
+        logger.info("Retrieved message: {}", message.getSubject());
+
+        // Get attachments if requested
+        java.util.List<io.kestra.plugin.microsoft365.outlook.domain.AttachmentInfo> attachmentInfos = new ArrayList<>();
+        if (rIncludeAttachment && Boolean.TRUE.equals(message.getHasAttachments())) {
+            logger.debug("Retrieving attachment information");
+
+            List<Attachment> attachments = GraphMailUtils.fetchAttachments(graphClient, rUser, rMsgId);
+            if (attachments == null || attachments.isEmpty()) {
+                logger.debug("No attachments found or failed to retrieve attachments");
+                attachments = new ArrayList<>();
+            }
+            attachmentInfos = attachments.stream()
+                .map(att -> io.kestra.plugin.microsoft365.outlook.domain.AttachmentInfo.builder()
+                    .id(att.getId())
+                    .name(att.getName())
+                    .contentType(att.getContentType())
+                    .size(att.getSize() != null ? att.getSize().longValue() : null)
+                    .build())
+                .toList();
+        }
+
+        io.kestra.plugin.microsoft365.outlook.domain.MessageDetail detail = io.kestra.plugin.microsoft365.outlook.domain.MessageDetail.builder()
```

Comment:
Just import it and remove the package 🙏 :

```suggestion
        MessageDetail detail = io.kestra.plugin.microsoft365.outlook.domain.MessageDetail.builder()
```